src/aiofiles/base.py

```diff
@@ -1,18 +1,54 @@
-from asyncio import get_running_loop
-from collections.abc import Awaitable
+from asyncio import get_running_loop, to_thread
+from collections.abc import Awaitable, Callable, Coroutine
 from contextlib import AbstractAsyncContextManager
-from functools import partial, wraps
+from functools import wraps
+from inspect import isfunction
+from warnings import warn
 
 
-def wrap(func):
+def to_coro(func: Callable):
+    """Converts the routine `func` into a coroutine.
+
+    The returned coroutine function runs the decorated function
+    in a separate thread.
+
+    Args:
+        func: A routine (regular function).
+
+    Returns:
+        A coroutine function.
+    """
+
     @wraps(func)
-    async def run(*args, loop=None, executor=None, **kwargs):
-        if loop is None:
-            loop = get_running_loop()
-        pfunc = partial(func, *args, **kwargs)
-        return await loop.run_in_executor(executor, pfunc)
+    async def _wrapper(*args, **kwargs) -> Coroutine:
+        return await to_thread(func, *args, **kwargs)
+
+    return _wrapper
+
+
+def asyncify(func: Callable):
+    """Returns an async version of the sync `func`.
+
+    Args:
+        func: A regular function.
+
+    Raises:
+        TypeError: If `func` is not callable.
+
+    Returns:
+        A coroutine function.
+    """
+
+    if isfunction(func):
+        return to_coro(func)
```

Comment:
If this PR goes first, then in the PR #193 the following extension may be added:

```python
if isgeneratorfunction(func):
    return to_agen(func)  # creates an async generator
```